model/src/plugin_handler.cpp

```diff
@@ -1180,22 +1199,32 @@ bool PluginHandler::ExtractMetadata(const std::string& path,
   std::string temp_path(tmpnam(0));
   if (!extractTarball(path, filelist, temp_path, true)) {
     std::ostringstream os;
-    os << "Cannot unpack plugin tarball at : " << path;
+    wxLogWarning("Cannot unpack plugin %s tarball at %s, check if tarball contains metadata.xml",
+      metadata.name.c_str(), path.c_str());
     if (filelist != "") cleanup(filelist, "unknown_name");
     last_error_msg = os.str();
     return false;
   }
-  if (!isRegularFile(temp_path.c_str())) return false;
+  if (!isRegularFile(temp_path.c_str())) {
+    // This could happen if the tarball does not contain the metadata.xml file
+    // or the metadata.xml file could not be extracted.
+    return false;
+  }
 
   struct CatalogCtx ctx;
   std::ifstream istream(temp_path);
   std::stringstream buff;
   buff << istream.rdbuf();
-  remove(temp_path.c_str());
-
+  int r = remove(temp_path.c_str());
+  if (r != 0) {
+    wxLogWarning("Cannot remove file %s: %s", temp_path.c_str(), strerror(r));
+  }
   auto xml = std::string("<plugins>") + buff.str() + "</plugins>";
   ParseCatalog(xml, &ctx);
   metadata = ctx.plugins[0];
+  if (metadata.name.empty()) {
+    wxLogWarning("Plugin metadata is empty");
```

Comment:
wxLogMessage, please.

Reply:
Are you ok if I change the code to display an error message indicating the "load plugin" operation failed, with a detailed error message? Currently, the "load plugin" operation fails silently.

About logging, it would be great to have guidelines when to use info, warning and error level.